### PR TITLE
[FIX] mail, *: others activities shown through the systray

### PR DESCRIPTION
**Pull request by bit-odoo**
**Current behavior before PR:**

Suppose we have a scenario like below,

- Task-A:
  - Activity-1:
       name: Email ( Today )
       assigned to: User-1

- Task-B:
  - Activity-1:
       name: Email ( Today )
       assigned to: User-2
    
  - Activity-2:
       name: Call ( Due in 3 Days )
       assigned to: User-1

Here, when user-1 go through the systray 'Today' filter shortcut, that time user
gets Task-A and Task-B both in the list instead of only Task-A.

Currently, activities are not filtered based on users with its deadlines.

**Desired behavior after PR is merged:**

Now activities will be filtered based on deadlines as well as the current user

To achieve this behavior we needed to pass the domain like,
[
   ('activity_ids.date_deadline','=', fields.Date.today()),
   ('activity_ids.user_id','=', 1)
]

And for that, we introduced a non-stored compute field with a search method.

**LINKS**
Task-2438822

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by Xavier-Do** [comment]
@KangOl, do you think  we could do something about that and avoid to catch computed fields. They are harmless most of the time.

**Comment by Xavier-Do** [comment]
Anyway @bit-odoo did someone already reviewed this task? do you know when this error was introdiced?

Could you also rework the commit message a little, it is a little difficult to understand. (And you don't need to add the pr number in the message mergebot should do that on merge ;)  )

**Comment by bit-odoo** [comment]
> Anyway @bit-odoo did someone already reviewed this task? do you know when this error was introdiced?
> 
> Could you also rework the commit message a little, it is a little difficult to understand. (And you don't need to add the pr number in the message mergebot should do that on merge ;) )

@Xavier-Do   Yes!! we made a master branch before this one  and @rco-odoo  has suggested this solution but as we are introducing  a non-stored compute field and its possible to do in stable thus we retargeted the branch in 14.0 and here that error introduced

update the commit message and PR you may check now.

**Comment by seb-odoo** [comment]
@bit-odoo what @Xavier-Do is mentioning is that you should wait for @tde-banana-odoo to review and approve this PR before requesting the runbot exception. Otherwise if it is not approved everyone wasted their time.

It's not a big deal if your runbot is red just because of that, you can ignore it until the final review.

File: addons/mail/models/mail_activity.py
Comment (tde-banana-odoo) [inline addons/mail/models/mail_activity.py:704]: Linting: would move this definition under 'activity_date_deadline' to ease model reading.

File: addons/mail/models/mail_activity.py
Comment (tde-banana-odoo) [inline addons/mail/models/mail_activity.py:705]: Don't forget other attributes, notably groups.

File: addons/mail/models/mail_activity.py
Comment (tde-banana-odoo) [inline addons/mail/models/mail_activity.py:762]: Technically also depends on activiyt_ids.user_id

File: addons/mail/models/mail_activity.py
Comment (tde-banana-odoo) [inline addons/mail/models/mail_activity.py:765]: That's strange as in multi mode, you link the first found activity to all records. Shouldn't you do it in a loop ?

File: addons/mail/models/mail_activity.py
Comment (tde-banana-odoo) [inline addons/mail/models/mail_activity.py:772]: Could be inlined in herebelow list

**Review by tde-banana-odoo — CHANGES_REQUESTED** [review]
Hello,

Globally LGTM. I added some side remarks, but approach seems good to me.

Could you also add some unit tests ? As we are adding new code in stable it should be tested accordingly :) .

Many thanks,

File: addons/mail/models/mail_activity.py
Comment (rco-odoo) [reply addons/mail/models/mail_activity.py:765]: I second that, you definitively need a loop on `self`.

File: addons/mail/models/mail_activity.py
Comment (tde-banana-odoo) [inline addons/mail/models/mail_activity.py:774]: We should also include a domain part about model being matching self, to avoid searching activities linked to other models.

File: addons/test_mail/tests/test_mail_activity.py
Comment (tde-banana-odoo) [inline addons/test_mail/tests/test_mail_activity.py:338]: I suggest to use existing data (users, test records) instead of re-creating everything here. It lessens boilerplate and allow to have smaller tests :) .

**Review by tde-banana-odoo — COMMENTED** [review]
Validation

File: addons/mail/models/mail_activity.py
Comment (hbh-odoo) [reply addons/mail/models/mail_activity.py:775]: @tde-banana-odoo Done!!

**Comment by tde-banana-odoo** [comment]
@robodoo r+

**Comment by tde-banana-odoo** [comment]
@robodoo r+

**Comment by tde-banana-odoo** [comment]
@Xavier-Do Mmmmh, the famous 14.0 branch that should have an exception, can't find what is wrong

**Comment by tde-banana-odoo** [comment]
@d-fence See hereabove message ^^

**Comment by d-fence** [comment]
Exception 54 was deactivated (we clean them once a week and this one was accidentally archived) + field `stock.inventory.my_activity_date_deadline` was missing.

The upgrade build was [red](https://runbot.odoo.com/runbot/build/8166895) because of the fastly failure on 2021-06-08.

Once merged, please forward port it to master asap without changes.

**Pull request closed without merging**
